unreal_asset/src/types/fname.rs

```diff
@@ -106,21 +106,35 @@ impl FName {
         FName::new_dummy(value.to_string(), 0)
     }
 
-    /// Get this FName content
-    pub fn get_content(&self) -> String {
-        // todo: return string ref
+    /// Get access to this `FName`'s content
+    pub fn get_content<T>(&self, func: impl FnOnce(&str) -> T) -> T {
         match self {
             FName::Backed {
-                index,
-                number: _,
-                ty: _,
-                name_map,
-            } => name_map.get_ref().get_name_reference(*index),
-            FName::Dummy { value, number: _ } => value.clone(),
+                name_map, index, ..
+            } => {
+                let name_map = name_map.get_ref();
+                func(name_map.get_name_reference(*index))
+            }
+            FName::Dummy { value, .. } => func(&value),
         }
     }
 
-    /// Get this FName instance number
+    /// Get this `FName`'s content as a `String`
+    pub fn get_owned_content(&self) -> String {
+        self.get_content(str::to_string)
+    }
+
+    /// Checks if an `FName`'s content is the given `&str`
+    pub fn is(&self, pat: impl AsRef<str>) -> bool {
```

Comment:
how about doing this via `ops::Eq`?

Reply:
ooh that's a good idea

Reply:
wait where is that? https://doc.rust-lang.org/std/ops/index.html

Reply:
oh, == is actually std::cmp

Reply:
👌